scripts/dataset/index.js

```diff
@@ -0,0 +1,55 @@
+import fs from 'fs';
+import path from 'path';
+import { fileURLToPath } from 'url';
+
+import generateRelease from './export/index.js';
+import logger from './logger/index.js';
+import publishRelease from './publish/index.js';
+
+const __dirname = path.dirname(fileURLToPath(import.meta.url));
+
+export async function release({ publicationEnabled, removeLocalCopyEnabled, fileName }) {
+  const releaseDate = new Date();
+  const archiveName = fileName || `dataset-${releaseDate.toISOString().replace(/T.*Z/, '')}`;
+  const archivePath = `${__dirname}/${path.basename(archiveName, '.zip')}.zip`; // Allow to pass both `filename` and `filename.zip` as args without having duplicated extension
+
+  const { servicesCount, firstCommitDate, lastCommitDate } = await generate({ archivePath, releaseDate });
+
+  if (!publicationEnabled) {
+    return;
+  }
+
+  return publish({ archivePath, releaseDate, servicesCount, firstCommitDate, lastCommitDate, publicationEnabled, removeLocalCopyEnabled });
+}
+
+export async function generate({ archivePath, releaseDate }) {
+  logger.info('Start exporting dataset…');
+
+  const { servicesCount, firstCommitDate, lastCommitDate } = await generateRelease({ archivePath, releaseDate });
+
+  logger.info(`Dataset exported in ${archivePath}`);
+
+  return { servicesCount, firstCommitDate, lastCommitDate };
+}
```

Comment:
4 lines of code, out of which 1 is to return a result and all variables are instantiated just for that. I suggest to bring it back to the main body 🙂

Reply:
It was to export it independently to the `publish` function but as we do not use it, I agree. 
I can also do the same for the `publish` function. Ok for that?

Reply:
I agree